### ISSUE TITLE
Incompatibility with Forgotten Armament 

### ISSUE THREAD

**Issue by neuroghast** [The-Gate-Project/DarkHorizons#32]
I've installed DH earlier in my list and then installed [Forgotten Armament by Morpheus.](https://www.morpheus-mart.com/forgotten-armament)

When installing the component **Update Items Setting an Ability Score to Increment the Increase** got the following error:

```
ERROR Installing [Update Items Setting an Ability Score to Increment the Increase], rolling back to previous state
Will uninstall  11 files for [FORGOTTEN-ARMAMENT/FORGOTTEN-ARMAMENT.TP2] component 4.
Uninstalled     11 files for [FORGOTTEN-ARMAMENT/FORGOTTEN-ARMAMENT.TP2] component 4.
ERROR: Failure("CMSTQ04.ITM: read out of bounds")
```
The item is from DH, an elemental staff.

Talking with Morpheus, he was able to find a solution for my case. Just notice this was his finding and solution, so any credits obviously goes to him. Pasting what he told me: "That item has a badly formed cast spell. For the magic ability, it does damage and doesn't cast a spell. Removing the magic ability solved the issue"

Updating the item, I was able to install the mod. Hope this helps!

**Comment by 11jo** [comment]
I am surprised that he didn't ask for my removal of the community because of it or simply put a total incompatibility between the two mods.


Anyway I will take a look at the item to see what's going wrong. Thanks for the report.

**Comment by GraionDilach** [comment]
This doesn't make any sense - all four elemental staves are set up the same way, so if that would be the issue, then the mod shouldn't have even reached CMSTQ04.ITM and should've crashed on CMSTQ02.ITM already. Besides, that item lacks any ability upgrade, so there's no reason for the component to even touch it!

Additionally, the last G3 download (if someone kept around that edition) doesn't crash on the file due to the above. This moreso sounds like an FA regression and misblamed on this mod.

**Comment by neuroghast** [comment]
Here's the log after successfullyi installing FA.

[WeiDU.log](https://github.com/user-attachments/files/18188761/WeiDU.log)

**Comment by GraionDilach** [comment]
> ~SETUP-DARKHORIZONSBGEE.TP2~ #0 #0 //  "Dark Horizons" Mod for Baldur's Gate: Enhanced Edition: BG:EE v213

You're not even using the TGP version!

**Comment by GraionDilach** [comment]
For the record, it is likely that this bug was fixed during TGP maintainance - likely within https://github.com/The-Gate-Project/DarkHorizons/commit/6a30ca4bcc0ef1af8628278b54a38b1bcbb5a430.

**Comment by 11jo** [comment]
@neuroghast,

Ok, indeed you seem to use a previous version of the mod.

This _particular_ item also [had un BWP fix in the past](https://github.com/The-Gate-Project/DarkHorizons/commit/c9865f5c252069f2f01e1697e95f80465e439ee9), so it's likely an old version issue.

Priorise the [latest release](https://github.com/The-Gate-Project/DarkHorizons/releases/tag/v3.01) for your futur installations.

Thanks @GraionDilach for looking into this _particular_ issue !

**Comment by GraionDilach** [comment]
Rechecked v213 - indeed, in that version, the dice number's missing from the 12 opcode, but that is properly fixed in TGP v3.01.
